apoc.meta.data count adoc/test (#2904)
neo4j-contrib/neo4j-apoc-procedures: docs/asciidoc/modules/ROOT/partials/usage/apoc.meta.data.adoc

Similarly, the `index` column show if there is an index or not, while the `existence` looks for an existence constraint.

The `array` column check if the row is of type array and, in case `type` columns is `"RELATIONSHIP"`,
the result will be `true` if there is at least one node with 2 outgoing relationships with the type of relation given by `label` or `property` column, 
so `ACTED_IN` in the example above.

The `leftCount`, `rightCount`, `left` and `right` columns regard only rows with column `type` equals to `"RELATIONSHIP"` (otherwise they are equal to `0`)
In particular, `leftCount`, `rightCount` are "weighted" counts of respectively outgoing and incoming relationships,
which for each node it will increase the count with the number of outgoing relations of that type for itself.
That is, if the node has 1 outgoing relationship with the provided type, the count will be increased by 1,
if it has 2 outgoing relationships, increased by 3, with 3 relationships, increased by 9, and so on.
Same thing for `rightCount`, but with incoming relationships instead.
So, with the above dataset, `(Person {name:'Keanu Reeves'})` has 5 relationships, so the count is 25,

> and `(:Person {name:'Tom Hanks'})` has 4 relationships, so the count is 16, this explain the total of 41.

Reviewer: I would write "... the total `leftCount `of 41." here to make it clearer

> Vice versa, every `(:Movie)` ha a single relationship, so the total il 9, like `count` result.

Reviewer: ```suggestion
Vice versa, every of the nine `(:Movie)` nodes has a single incoming relationship, so the total `rightCount`result is 9.
```

Something like this maybe?

.Results
[opts="header"]
|===
| label        | property     | count | unique | index | existence | type           | array | sample | left | right | other                          | otherLabels | elementType
| "ACTED_IN"   | "Person"     | 4     | false  | false | false     | "RELATIONSHIP" | true  | null   | 4    | 0     | ["Movie"]                      | []          | "relationship"
| "ACTED_IN"   | "roles"      | 0     | false  | false | false     | "LIST"         | true  | null   | 0    | 0     | []                             | []          | "relationship"

> | "ACTED_IN"   | "Another"    | 1     | false  | false | false     | "RELATIONSHIP" | true  | null   | 5    | 0     | ["Movie"]                      | []          | "relationship"

Author: Yes, sorry. Something wrong it's remained during the various changes, changed

|===

The `unique` column shows if there is an unique constraint in that specific label and property.
Similarly, the `index` column show if there is an index or not, while the `existence` looks for an existence constraint.

The `array` column check if the row is of type array and, in case `type` columns is `"RELATIONSHIP"`,
the result will be `true` if there is at least one node with 2 outgoing relationships with the type of relation given by `label` or `property` column, 
so `ACTED_IN` in the example above.

The `left`, `right` and `count` columns regard only rows with column `type` equals to `"RELATIONSHIP"` (otherwise they are equal to `0`).
Please note that, because we examine a sample, these counts are just estimates to give an overview and proximity to actual values depends on the dataset and the sample set (default `100`).

In particular the `count` column indicates the number of nodes with an outgoing relationship 
(e.g. the row with label = `ACTED_IN` and property = `Person` has `count` 2 because there are 2 nodes `(node:Person)-[:ACTED_IN]->()`, i.e. `(Keanu)` and `(TomH)`).

> The `left` value represents the ratio (rounded down) of the sum of outgoing node-degrees to `count`, for each relationship.

Author: I added the corresponding cypher statement, to be clearer. Let me know if it can be ok

> The `left` value represents the ratio (rounded down) of the sum of outgoing node-degrees to `count`, for each relationship.

Reviewer 2: That is great, thank you :smile:

> For example, regarding the row with label = `RELATED_TO` and property = `Movie`, has an outgoing node-degree equals to `1` ,because there is one relationship `(:Movie)-[rel:RELATED_TO]->()`, i.e. `(TheMatrix)-[:RELATED_TO {rate:34}]->(Laurence)` and `(TheMatrixReloaded)-[:RELATED_TO {rate:345}]->(LanaW)`.

Reviewer 2: ```suggestion
For example, the row with label = `RELATED_TO` and property = `Movie` has an outgoing node-degree equal to `1`, because there is one relationship `(:Movie)-[rel:RELATED_TO]->()`, i.e. `(TheMatrix)-[:RELATED_TO {rate:34}]->(Laurence)` and `(TheMatrixReloaded)-[:RELATED_TO {rate:345}]->(LanaW)`.
```
